build.gradle

```diff
@@ -1,27 +1,18 @@
 buildscript {
     repositories {
-        maven {
-            url "https://research.rightmesh.io/artifactory/libs-local"
-            credentials {
-                username artifactory_app_username
-                password artifactory_app_password
-            }
-        }
-    }
-    dependencies {
-        classpath 'io.left.rightmesh:rightmesh-plugin:1.6'
+        mavenCentral()
+        jcenter()
     }
 }
-
 plugins {
     id 'application'
+    id 'io.left.rightmesh.rightmesh-plugin' version '1.8.1-SNAPSHOT'
 }
 
-apply plugin: 'io.left.rightmesh.rightmesh-plugin'
-build.dependsOn("rightmesh")
-
 rightmesh {
-    outputs.dir file("src/main/assets/")
+    appKey = rightmesh_hellojavamesh_key
+    username = rightmesh_devportal_username
```

Comment:
I kind of like the idea of encouraging developers to specify exactly which variables there buildscripts are referencing, including encouraging setting specific key variables for each app. That will hopefully make our onboarding process easier, and reduce credential conflicts (where we have to comment out credentials temporarily to compile different apps).

Reply:
The app key is different for each app, but isn't our flow that a developer's credentials will be the same for all of them? And at that point shouldn't we just encourage using the default values?

Reply:
They aren't always; artifactory-internal and artifactory credentials can be different. Not to mention, the "default values" are just an arbitrary variable name. It makes sense to me to get rid of those completely once everyone is in the habit of specifying their credentials within the `rightmesh { }` block. The more explicit we can be, the less confusion there will hopefully be going forward.

Reply:
Having default values and providing methods to override them is a common pattern (e.g., system-wide git author information, standard filenames like Makefile/Dockerfile or .gitignore, default environment paths). I would be against removing it.

Would your intention be to use the same `rightmesh_devportal_username` pattern in all of the public apps?

Reply:
I suggest using rightmesh_devportal_username, explicitly specified. Having defaults is good too, but a default variable of artifactory_app_username seems non-descriptively named for a system-wide default that is only used by rightmesh apps. If we keep a system-wide default, I propose both encouraging explicitly mapping that variable into the rightmesh block, and renaming artifactory_app_username to rightmesh_devportal_username (or something similarly descriptive of its intended scope). For the time being, we preserve the existing default of artifactory_app_username only for backwards compatibility while we update our sample apps.

Reply:
I agree. :+1: That all sounds good to me.

"Should we update the default name to match this?" was going to be my next question. I'm fine with encouraging being specific, I'm just hoping to avoid having to define a whole bunch of credentials in `gradle.properties`, so if we're at least consistent with naming it'll save us a little bit of sanity.

Reply:
I agree. +1 That all sounds good to me.

"Should we update the default name to match this?" was going to be my next question. I'm fine with encouraging being specific, I'm just hoping to avoid having to define a whole bunch of credentials in gradle.properties, so if we're at least consistent with naming it'll save us a little bit of sanity.